cwa/olin/src/lib.rs

```diff
@@ -1,54 +1,92 @@
+/// Main classes to perform the low-level operations
+/// described in the Olin specs. In particular the basic
+/// "actions": open, read, write, close, flush on files.
+/// 
+/// Modules:
+/// * sys
+/// * err
+/// * log
+/// * env
+/// * runtime
+/// * startup
+/// * time
+/// * stdio
+/// * random
+///
+/// ```Resource``` class defines a generic class for a resource
+/// with methods bound for FFI access
+
 extern crate chrono;
 
 use std::io::{self, Read, Write};
 
+/// import HTTP client from local
+/// TODO: couldn't get how this is used as it seems not to be called anywhere
 pub mod http;
+/// import panic handling from local
 pub mod panic;
 
+/// system-level operations, interface and FFI access:
+/// * setup and low-level operations
+/// * bindings to environment variables
+/// * bindings for Go runtime
```

Comment:
None of the comments for these functions help as is. I would suggest linking to https://github.com/CommonWA/cwa-spec for the individual functions in question.

Reply:
this link should be in a README in the `cwa` directory